File: simple_linear_regression.r
#SIMPLE LINEAR REGRESSION

#IMPORTING THE DATASET
dataset=read.csv("D:\\DATA  SCIENCE\\MACHINE LEARNING\\SUPERVISED LEARNING\\SIMPLE LINEAR REGRATION-MODEL\\Project1-simple linear regression\\Salary_Data.csv")
print(dataset)
fix<-(dataset)
fit<-lm(dataset$Salary~dataset$YearsExperience,data=dataset)
summary(fit)

#splitting the dataset into the training set and test set
install.packages('caTools')
library(caTools)
set.seed(123)
split=sample.split(dataset$Salary, SplitRatio = 2/3)
training_set=subset(dataset,split==TRUE)
test_set=subset(dataset,split==FALSE)

#FEATURE SCALING
#training_set=scale(training_set)
#test_set=scale(test_set)

#FITTING SIMPLE LINEAR REGRESSION TO THE TRAINING SET
regressor= lm(formula=Salary ~YearsExperience, data=training_set)
summary(regressor)

#predicting the test set results
y_pred=predict(regressor,dataset = test_set)
print(y_pred)
head(x)
#visualising the training set results
install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Training set)') +
  xlab('Years of experience') +
  ylab('Salary')

# Visualising the Test set 
library(ggplot2)
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Test set)') +
  xlab('Years of experience') +
  ylab('Salary')

  

File: simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Mon Mar 11 11:07:28 2019

@author: dell
"""
#import the libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

#Read the data set
data=pd.read_csv("D:\DATA  SCIENCE\MACHINE LEARNING\SUPERVISED LEARNING\SIMPLE LINEAR REGRATION-MODEL\Project1-simple linear regression\salary.csv")

#divide  the dataset in to x and y
X=data.iloc[:,:-1].values
y=data.iloc[:,1].values

#splitingb the  data the data based on trending and test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

#implemente our classifier based on simple linear regression
from sklearn.linear_model import LinearRegression
simplelinearRegression=LinearRegression()
simplelinearRegression.fit(X_train,y_train)

y_predict=simplelinearRegression.predict(X_test)


#implement the graph
plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train, simplelinearRegression.predict(X_train))
plt.show()

# Visualising the Test set results
plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_train, simplelinearRegression.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()